edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/device/NestingInfo.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Core.Device
+{
+    using System;
+    using System.Threading;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class NotNested : INestingInfo
+    {
+        public static INestingInfo Instance { get; } = new NotNested();
+
+        public Option<IIdentity> KnownParent => Option.None<IIdentity>();
+        public bool IsDirectClient => true;
+
+        public void BindToParent(IIdentity parent) => throw new InvalidOperationException("Nesting info is not supported");
+    }
+
+    public class NestingInfo : INestingInfo
+    {
+        // Using null because BindToParent is implement with interlocked operations and that does not work with value types (like Option)
+        IIdentity knownParent = null;
```

Comment:
:) you can also use option and lock in bind.

Reply:
I already feel bad for a synchronized just to give a warning that could not happen in theory, would feel even worse locking